docs/types/reference-types.rst

```diff
@@ -550,3 +550,9 @@ members of the local variable actually write to the state.
 Of course, you can also directly access the members of the struct without
 assigning it to a local variable, as in
 ``campaigns[campaignID].amount = 0``.
+
+.. note::
+    In previous solidity versions, assigments which create memory-structs that contain a mapping
+    like ``campaigns[campaignID] = Campaign(beneficiary, goal, 0, 0)`` were possible,
+    because the compiler would omit the types, in this case the mapping, which are not 
+    implicitly convertible to ``memory``.
```

Comment:
To be perfectly clear, the assignment behavior was just a consequence of how such structs worked in memory. I suggest making this not a bit more general for that reason. Here's how I would phrase it:

```suggestion
    In previous Solidity versions, memory-structs containing members of storage-only types (e.g. mappings),
    were allowed and assignments like ``campaigns[campaignID] = Campaign(beneficiary, goal, 0, 0)``
    in the example above would work and just silently skip those members.
```

Reply:
Yeah, I think it sounds much better like this + https://github.com/ethereum/solidity/pull/10431#discussion_r532090955